### PR TITLE
Fix ComplexTypeMark handling for pattern from QwenImage

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
- Fix model conversion for Qwen-Image with complex buffers
- Add `prim::TupleUnpack` and `aten::type_as` translators with complex tensor support
- Propagate ComplexTypeMark through `prim::ListUnpack` and `prim::GetAttr` operations
- Move `PrimListUnpackReplacer` to no_val pass block

### Tickets:
 - 176305

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_listunpack.py [hunk 1]
```diff
@@ -322,3 +322,36 @@ def forward(self, x: list[torch.Tensor], y: list[torch.Tensor]):
...
+
+class TestComplexListUnpack(PytorchLayerTest):
+    """Test prim::ListUnpack with complex tensors (ComplexTypeMark propagation).
+
+    This tests the changes in list_unpack.cpp that handle ComplexTypeMark
+    propagation through ListUnpack operations.
+    """
+
+    def _prepare_input(self):
+        return (np.random.randn(2, 4, 2).astype(np.float32),)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_listunpack.py]: Use deterministic input:
```
rng = np.random.default_rng(43)
self.input_tensor = rng.normal(2, 4, 2).astype(np.float32),)

File: tests/layer_tests/pytorch_tests/test_listunpack.py [hunk 2]
```diff
@@ -322,3 +322,36 @@ def forward(self, x: list[torch.Tensor], y: list[torch.Tensor]):
...
+
+    def create_model(self):
+        class ComplexListUnpack(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                freqs = torch.randn(4, 2)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_listunpack.py]: Use deterministic input:
```
g = torch.Generator().manual_seed(42)
freqs = torch.randn(4, 2, generator=g)

File: tests/layer_tests/pytorch_tests/test_rope_pattern.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    def _prepare_input(self, seq_len=8, dim=64):
+        return (np.random.randn(1, seq_len, dim).astype(np.float32),)
+
+    def create_model(self, dim=64):
+        class RoPEModule(torch.nn.Module):
+            def __init__(self, dim):
+                super().__init__()
+                # Create freqs_cis as complex buffer (like in Qwen-Image)
+                freqs = torch.randn(8, dim // 2)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_rope_pattern.py]: Use deterministic input:
```
g = torch.Generator().manual_seed(42)
freqs = torch.randn(8, dim // 2, generator=g)

File: tests/layer_tests/pytorch_tests/test_rope_pattern.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    """
+
+    def _prepare_input(self):
+        return (np.random.randn(1, 4, 8).astype(np.float32),)
+
+    def create_model(self):
+        class ComplexBufferUnsqueeze(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                real = torch.randn(4, 8)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_rope_pattern.py]: Use deterministic input (see snippets in similar comments).

File: tests/layer_tests/pytorch_tests/test_rope_pattern.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    """
+
+    def _prepare_input(self):
+        return (np.random.randn(1, 4, 2).astype(np.float32),)
+
+    def create_model(self):
+        class ComplexBufferMultipleOps(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                freqs = torch.randn(4, 2)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_rope_pattern.py]: Use deterministic input (see snippets in similar comments).

File: tests/layer_tests/pytorch_tests/test_rope_pattern.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+
+class TestComplexBufferMultipleOps(PytorchLayerTest):
+    """Test complex buffer with multiple operations.
+
+    Tests complex buffer going through multiple operations before view_as_real.
+    """
+
+    def _prepare_input(self):
+        return (np.random.randn(1, 4, 2).astype(np.float32),)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_rope_pattern.py]: Use deterministic input (see snippets in similar comments).

File: tests/layer_tests/pytorch_tests/test_tuple_unpack.py [hunk 7]
```diff
@@ -32,3 +34,78 @@ def forward(self, input_data):
...
+
+    def _prepare_input(self):
+        return (np.random.randn(2, 4, 2).astype(np.float32),)
+
+    def create_model(self):
+        class ComplexTupleUnpack(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                # Use separate buffers instead of tuple
+                freqs = torch.randn(4, 2)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_tuple_unpack.py]: Use deterministic input (see snippets in similar comments).

File: tests/layer_tests/pytorch_tests/test_tuple_unpack.py [hunk 8]
```diff
@@ -32,3 +34,78 @@ def forward(self, input_data):
...
+    - src/frontends/pytorch/src/op/tuple_unpack.cpp
+    - src/frontends/pytorch/src/transforms/tuple_unpack_replacer.cpp
+
+    The test verifies that complex tensor operations work correctly when
+    tensors are used in tuple-like patterns. We use trace_model=True to
+    get proper TorchScript representation.
+    """
+
+    def _prepare_input(self):
+        return (np.random.randn(2, 4, 2).astype(np.float32),)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_tuple_unpack.py]: Use deterministic input (see snippets in similar comments).

File: tests/layer_tests/pytorch_tests/test_tuple_unpack.py [hunk 9]
```diff
@@ -32,3 +34,78 @@ def forward(self, input_data):
...
+    """Test TupleUnpack with multiple complex tensor outputs."""
+
+    def _prepare_input(self):
+        return (np.random.randn(2, 4, 2).astype(np.float32),)
+
+    def create_model(self):
+        class ComplexTupleUnpackMultiple(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                freqs = torch.randn(4, 2)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_tuple_unpack.py]: Use deterministic input (see snippets in similar comments).

File: tests/layer_tests/pytorch_tests/test_tuple_unpack.py [hunk 10]
```diff
@@ -32,3 +34,78 @@ def forward(self, input_data):
...
+    def test_complex_tuple_unpack(self, ie_device, precision, ir_version):
+        self._test(*self.create_model(), ie_device, precision, ir_version,
+                   trace_model=True)
+
+
+class TestComplexTupleUnpackMultiple(PytorchLayerTest):
+    """Test TupleUnpack with multiple complex tensor outputs."""
+
+    def _prepare_input(self):
+        return (np.random.randn(2, 4, 2).astype(np.float32),)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_tuple_unpack.py]: Use deterministic input (see snippets in similar comments).

File: tests/layer_tests/pytorch_tests/test_type_as.py [hunk 11]
```diff
@@ -32,3 +32,34 @@ def forward(self, x, y):
...
         self._test(*self.create_model(), ie_device, precision, ir_version,
                    kwargs_to_prepare_input={"input_dtype": input_dtype, "cast_dtype": cast_dtype})
+
+
+class TestComplexTypeAs(PytorchLayerTest):
+    """Test aten::type_as with complex tensors."""
+
+    def _prepare_input(self, input_dtype=np.float32, cast_dtype=np.float64):
+        # Complex tensor represented as real with shape [..., 2]
+        return (np.random.randn(2, 4, 2).astype(input_dtype),
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_type_as.py]: Use deterministic input (see snippets in similar comments).

**Comment by mvafin** [comment]
Agree with @bumbosiepsak, we need to fix the seed in the test
